chia/rpc/wallet_rpc_client.py

```diff
@@ -374,16 +377,18 @@ async def get_cat_asset_id(self, wallet_id: str) -> bytes:
         }
         return bytes.fromhex((await self.fetch("cat_get_asset_id", request))["asset_id"])
 
-    async def cat_asset_id_to_name(self, asset_id: bytes32) -> Optional[Tuple[uint32, str]]:
+    async def cat_asset_id_to_name(self, asset_id: bytes32) -> Optional[Tuple[Optional[uint32], str]]:
         request: Dict[str, Any] = {
             "asset_id": asset_id.hex(),
         }
         try:
             res = await self.fetch("cat_asset_id_to_name", request)
-            return uint32(int(res["wallet_id"])), res["name"]
-        except Exception:
+        except ValueError:
             return None
 
+        wallet_id: Optional[uint32] = None if res["wallet_id"] is None else uint32(int(res["wallet_id"]))
+        return wallet_id, res["name"]
```

Comment:
what the purpose of returning wallet_id here ?

Reply:
`cat_asset_id_to_name` is called by `add_token` in wallet_funcs.py.

It's used to get the wallet_id so that the wallet can be renamed:
`await wallet_client.set_cat_name(wallet_id, token_name)`